config/v1/types_oauth.go

```diff
@@ -1,37 +1,420 @@
+	Templates *OAuthTemplates `json:"templates"`
 }
 
 type OAuthStatus struct {
+	operatorsv1alpha1api.OperatorStatus `json:",inline"`
+}
+
+// OAuthTemplates allow for customization of pages like the login page
+type OAuthTemplates struct {
+	// Login is a path to a file containing a go template used to render the login page.
+	// If unspecified, the default login page is used.
+	Login string `json:"login"`
+
+	// ProviderSelection is a path to a file containing a go template used to render the provider selection page.
+	// If unspecified, the default provider selection page is used.
+	ProviderSelection string `json:"providerSelection"`
+
+	// Error is a path to a file containing a go template used to render error pages during the authentication or grant flow
+	// If unspecified, the default error page is used.
+	Error string `json:"error"`
+}
+
+// GrantHandlerType are the valid strategies for handling grant requests
+type GrantHandlerType string
+
+const (
+	// GrantHandlerAuto auto-approves client authorization grant requests
+	GrantHandlerAuto GrantHandlerType = "auto"
+	// GrantHandlerPrompt prompts the user to approve new client authorization grant requests
+	GrantHandlerPrompt GrantHandlerType = "prompt"
+	// GrantHandlerDeny auto-denies client authorization grant requests
+	GrantHandlerDeny GrantHandlerType = "deny"
+)
+
+// GrantConfig holds the necessary configuration options for grant handlers
+type GrantConfig struct {
+	// Method: allow, deny, prompt
+	Method GrantHandlerType
+
+	// ServiceAccountMethod is used for determining client authorization for service account oauth client.
+	// It must be either: deny, prompt
+	ServiceAccountMethod GrantHandlerType
+}
+
+var ValidGrantHandlerTypes = sets.NewString(string(GrantHandlerAuto), string(GrantHandlerPrompt), string(GrantHandlerDeny))
+var ValidServiceAccountGrantHandlerTypes = sets.NewString(string(GrantHandlerPrompt), string(GrantHandlerDeny))
+
+// TokenConfig holds the necessary configuration options for authorization and access tokens
+type TokenConfig struct {
+	// AuthorizeTokenMaxAgeSeconds defines the maximum age of authorize tokens
+	AuthorizeTokenMaxAgeSeconds int32 `json:"authorizeTokenMaxAgeSeconds"`
+	// AccessTokenMaxAgeSeconds defines the maximum age of access tokens
+	AccessTokenMaxAgeSeconds int32 `json:"accessTokenMaxAgeSeconds"`
+	// AccessTokenInactivityTimeoutSeconds defined the default token
+	// inactivity timeout for tokens granted by any client.
+	// Setting it to nil means the feature is completely disabled (default)
+	// The default setting can be overriden on OAuthClient basis.
+	// The value represents the maximum amount of time that can occur between
+	// consecutive uses of the token. Tokens become invalid if they are not
+	// used within this temporal window. The user will need to acquire a new
+	// token to regain access once a token times out.
+	// Valid values are:
+	// - 0: Tokens never time out
```

Comment:
@smarterclayton I think my preference in cases like this is negative values are "never time out", zero means disabled, positive is that amount.  That is consistent with "normal" libraries like [`strings.Replace`](https://golang.org/pkg/strings/#Replace)

Reply:
thinking about it from the perspective of a JSON client that might be using this API, distinguishing between nil and 0 in javascript requires a `===` and is a headache waiting to happen. Whereas there's nothing ambiguous about negative number versus 0. So I think I agree with @deads2k on this one.

Reply:
I agree, that's more consistent. changed in a9effe1